forum/users/permissions.py

```diff
@@ -0,0 +1,37 @@
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+from rest_framework.exceptions import PermissionDenied
+from rest_framework_simplejwt.tokens import AccessToken
+from .models import Role
+
+class BaseRolePermission(BasePermission):
+    """
+    Permission class to check if a user has the required role,
+    based on both the token and the database state.
+    """
+    required_role: Role
+
+    def has_permission(self, request: Request, view) -> bool:
+        if not request.user.is_authenticated:
+            return False
+
+        if not isinstance(request.auth, AccessToken) or request.auth is None:
+            raise PermissionDenied("Something is wrong with auth: user's probably not using JWT or another problem")
+
+        token_role = request.auth.get('role')
+
+        if self.required_role == Role.UNASSIGNED:
+            return token_role == Role.UNASSIGNED.value
+
```

Comment:
Use dynamic checks for better scalability and maintainability. For example, encapsulate role validation logic within a helper method:
```
def is_role_valid(token_role, required_role):
    return token_role == required_role.value
```
Replace direct comparisons with calls to this function:
```
if self.required_role == Role.UNASSIGNED:
    return is_role_valid(token_role, Role.UNASSIGNED)
```

Reply:
Done ✅